### PR TITLE
More rework of GitHub module structure

### PR DESCRIPTION
**Pull request by benbc**
There is a point to all this, I promise. Next PR will have actual functional changes.

### CODE REVIEW

**Review by lucyb — APPROVED** [review]
This is a significant improvement and, as ever, a lovely set of commits to read through. It's been a pleasure to see the code base evolve and it's looking really nice now.

**Review by iaindillingham — APPROVED** [review]
FWIW, I found this PR hardest of the recent PRs to review. I feel I need to finish reading _Domain Driven Design_ to truly understand this PR.

File: metrics/github/security.py [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 from dataclasses import dataclass
-from datetime import date
 
 from ..tools import dates
-from . import query, repos
+from . import github, query
 
 
 @dataclass
 class Vulnerability:
```
[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/security.py:9]: Yes, I considered it. I decided that I'd limit the refactoring to getting the bits of domain logic I _needed_ in one place for my forthcoming change (using team membership to handle some tech/non-tech PRs differently).

I can see an opportunity coming to rework the vulnerability stuff as part of the next stage of dependabot/security work.

File: metrics/github/query.py [hunk 2]
```diff
@@ -132,15 +124,24 @@ def issues(client, repo):
...
+            org=org,
+            repo=repo,
         )
     )
 
 
+def _client():
+    return GitHubClient(
+        tokens={
+            "ebmdatalab": os.environ["GITHUB_EBMDATALAB_TOKEN"],
```
[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/query.py:139]: I know. I would almost always prefer to read env vars at the top level. I just got fed up passing the `client` around.

An alternative that occurs to me now is that we could read the env vars at the top level and then set a property on the `client` (or `query`?) module with the token. Maybe that's gross.

[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/query.py:139]: I get the impression that you don't _hate_ this as it is, so I'll merge the PR and we can reconsider later.

**Pull request merged**